### DISCUSSION TITLE
Is it possible to export single email instead of all of them?

### DISCUSSION THREAD

**Discussion by minikolic** [resend/react-email#1589] [Ideas]
### Goals

1. I would like to export a single email by passing the additional argument to the `export` command

### Background

Our system has hundreds of emails, and exporting them takes some time to complete. It would be great if additional flags can be passed to the `export` command to export only the given email instead of all of them. It would be a huge time saver.

### Proposal

something like this should work:
`export --email my-cool-email.tsx` 
`export --email my-cool-email.tsx --email my-other-cool-email.tsx`

**Comment by gabrielmfern** [comment]
Why exactly would it be a time saver? Is it because you are doing `email export` consecutive times while doing consecutive changes to emails? If that's the case, could an `email export --watch` do it better for you here?

**Reply by minikolic** [reply]
Nope, we are using a live preview during the development phase, which fulfills our needs for most cases. But invoking `export` command (usually on the Github Action) takes ~5 mins to complete. And we are expecting this time to grow, as soon as we add more emails to the system. We are using HTML files for two cases:

* We are uploading HTMLs to the external mail system that handles email sending. In many cases, we upload just a few html files per run, so we would like to have an optimized `export` command, that will export just the necessary things.

* During the developing phase, sometimes, we pass HTMLs to the Thymeleaf engine that uses given files to validate exported HTMLs as we are injecting Thymeleaf expressions into the emails. For this validation, we would also like to export just one or few emails

**Comment by ryanoglesby08** [comment]
+1 I'd also be interested in this feature.

My company has built this functionality on top of react-email, and exporting by a file path is something we wanted but had to hand-roll. We would love to be able to just run `export` directly to do it. It would seem logical, as  `export` is described in the react-email docs as a secondary rendering method to `render`. And with `render` you can specify a single template with `render(<Email />)`.

Exporting by file path is useful because it is a more targeted rendering process. If email "A" has a syntax error or something, that shouldn't stop me from exporting email "B"

**Comment by multivoltage** [comment]
Docs say to use `export` is you cannot used `render`. Since my company has got 100 mail, my build takes maybe 6 seconds so is good for me. But since `export` generate the three structure in a different way than I wanted, I created a script for my custom export like:

```js
function export(){
  fs.readdirSync("./emails").filter((folderName) => {
    const thereisFilter = getFilteredFromProcessArgv() 
    return isEmailFolder(folderName) && folderName.startsWith(thereisFilter)
  }).forEach(folderName => {
    const module = await import(`./emails/mail.tsx`);

    const html = await render(
      <module.default {...module.default.PreviewProps} />,
      {
        ...options,
        plainText: false,
      }
    );
    fs.writeFileSync("output.html",{ encoding: "utf-8"})
  })
}
```

This is a simplified version of my code. But the idea is the same. You can pass some `--mailName=hello` in package.json is you want
This script is very fast under node 20 in my case for 100 mail :). And you it will takes too much time you can pay with threads.
